crates/resource-resolver/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+    // better alternative
+    if cfg!(CARGO_PACKAGER_FORMAT = "app") {
+        PackageFormat::App
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "dmg") {
+        PackageFormat::Dmg
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "wix") {
+        PackageFormat::Wix
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "nsis") {
+        PackageFormat::Nsis
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "deb") {
+        PackageFormat::Deb
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "appimage") {
+        PackageFormat::AppImage
+    } else {
+        PackageFormat::None
+    }
+}
+
+/// Retreive the resource path of your app, packaged with cargo packager.
+/// This function behave the same as [`resource_dir`], except it accepts
+/// a parameter that will be happened to the resource path when no packaging format
+/// is used.
+///
+/// Example: You want to include the folder `crate/resource/icons/`.
+///
+/// - With `cargo run` command, you will have to execute
+///     `resource_dir().unwrap().join("resource/icons/")` to get the path.
+/// - With any other formats, it will be `resource_dir().unwrap().join("icons/")`.
+///
+/// ```
+/// use cargo_packager_resource_resolver::{resolve_resource, PackageFormat};
+///
+/// resolve_resource(PackageFormat::None, "resource").unwrap().join("icons/");
+/// ```
+pub fn resolve_resource<P: AsRef<Path>>(package_format: PackageFormat, path: P) -> Result<PathBuf> {
+    match package_format {
+        PackageFormat::None => {
+            let root_crate_dir = env::var("CARGO_MANIFEST_DIR")
+                .map_err(|e| {
+                    match e {
+                        env::VarError::NotPresent => {
+                            Error::Env("PackageFormat::None was use, but CARGO_MANIFEST_DIR environnement variable was not defined".to_string())
+                        },
+                        _ => Error::Var("Can't access CARGO_MANIFEST_DIR environnement variable".to_string(), e)
+                    }
+                })?;
+            Ok(PathBuf::from(root_crate_dir).join(path))
+        }
+        PackageFormat::App | PackageFormat::Dmg => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            exe_dir
+                .join("../Resources")
+                .canonicalize()
+                .map_err(|e| Error::Io("".to_string(), e))
+        }
+        PackageFormat::Wix | PackageFormat::Nsis => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            Ok(exe_dir.to_path_buf())
+        }
+        PackageFormat::Deb => {
+            // maybe this is not reliable, and we need to get the app name from argument
+            let exe = current_exe()?;
+            let binary_name = exe.file_name().unwrap().to_string_lossy();
+
+            let path = format!("/usr/lib/{}/", binary_name);
+            Ok(PathBuf::from(path))
+        }
+        PackageFormat::AppImage => Err(Error::UnsupportedPlatform),
```

Comment:
tauri supports it: https://github.com/tauri-apps/tauri/blob/97e334129956159bbd60e1c531b6acd3bc6139a6/core/tauri-utils/src/platform.rs#L226